src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -77,7 +77,7 @@ public class XmlLoggingConfiguration : LoggingConfigurationParser, IInitializeSu
         /// Initializes a new instance of the <see cref="XmlLoggingConfiguration" /> class.
         /// </summary>
         /// <param name="fileName">Configuration file to be read.</param>
-        public XmlLoggingConfiguration([CanBeNull] string fileName)
+        public XmlLoggingConfiguration([NotNull] string fileName)
```

Comment:
Cannot see any parameter called XML-string in that constructor. This constructor can only load from filename.

Reply:
the commit is one the wrong line I see, I mean this:

```
            return new XmlLoggingConfiguration(XmlReader.Create(new StringReader($"<nlog autoReload='{autoReload}'></nlog>")), configFile, logFactory);    // Empty default config, but monitors file
```

Reply:
For some reason we cannot provide a LogFactory to the constructor that takes XML-string. So I just use what we have.

Reply:
I mean, why not:

```c#
  var config = new XmlLoggingConfiguration(configFile,logFactory) { AutoReload = autoReload};
```

what am I missing?

Reply:
You have just tried loading the file and failed, because the xml-parser blew up in your face. But we can see that autoreload has been configured by looking at filecontent. There is little point in trying to load the file again, so instead we just load dummy-content and wait for the file to be updated.

Reply:
ow yes we need to create a XmlLoggingConfiguration with that filename, but skip the parse.

Still I think there should be a better way (internal? property or constructor), then parse a string.

Reply:
or what about a XmlLoggingConfiguration.CreateDummy(finename, autoreload, factory). Make it the issue of XmlLoggingConfiguration classs (and I hope we don't need the parse then)

Reply:
Like the idea with an internal constructor, that can become public when people screams for it.

Reply:
Done